packages/react-router/tests/router.test.tsx

```diff
@@ -2836,6 +2836,72 @@ describe('rewriteBasepath utility', () => {
     expect(router.state.location.pathname).toBe('/test')
   })
 
+  it('should handle basepath when accessing root path and maintain basepath in browser URL', async () => {
+    const rootRoute = createRootRoute({
+      component: () => <Outlet />,
+    })
+
+    const indexRoute = createRoute({
+      getParentRoute: () => rootRoute,
+      path: '/',
+      component: () => <div data-testid="home">Home</div>,
+    })
+
+    const routeTree = rootRoute.addChildren([indexRoute])
+
+    const history = createMemoryHistory({
+      initialEntries: ['/my-app/'],
+    })
+
+    const router = createRouter({
+      routeTree,
+      history,
+      rewrite: rewriteBasepath({ basepath: '/my-app' }),
```

Comment:
please go back to `basepath` prop for testing this

we reverted the deprecation of that prop